projects/packages/my-jetpack/src/products/class-social.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * Search product
+ *
+ * @package my-jetpack
+ */
+
+namespace Automattic\Jetpack\My_Jetpack\Products;
+
+use Automattic\Jetpack\My_Jetpack\Hybrid_Product;
+use Automattic\Jetpack\My_Jetpack\Wpcom_Products;
+
+/**
+ * Class responsible for handling the Social product
+ */
+class Social extends Hybrid_Product {
+
+	/**
+	 * The product slug
+	 *
+	 * @var string
+	 */
+	public static $slug = 'social';
+
+	/**
+	 * The Jetpack module name
+	 *
+	 * @var string
+	 */
+	public static $module_name = 'social';
+
+	/**
+	 * The slug of the plugin associated with this product.
+	 *
+	 * @var string
+	 */
+	public static $plugin_slug = 'jetpack-social';
+
+	/**
+	 * The filename (id) of the plugin associated with this product.
+	 *
+	 * @var string
+	 */
+	public static $plugin_filename = array(
+		'jetpack-social/jetpack-social.php',
+		'social/jetpack-social.php',
+		'jetpack-social-dev/jetpack-social.php',
+	);
+
+	/**
+	 * Get the internationalized product name
+	 *
+	 * @return string
+	 */
+	public static function get_name() {
+		return __( 'Social', 'jetpack-my-jetpack' );
+	}
+
+	/**
+	 * Get the internationalized product title
+	 *
+	 * @return string
+	 */
+	public static function get_title() {
+		return __( 'Jetpack Social', 'jetpack-my-jetpack' );
+	}
+
+	/**
+	 * Get the internationalized product description
+	 *
+	 * @return string
+	 */
+	public static function get_description() {
+		return __( 'Reach your audience on social media', 'jetpack-my-jetpack' );
+	}
+
+	/**
+	 * Get the internationalized product long description
+	 *
+	 * @return string
+	 */
+	public static function get_long_description() {
+		return __( 'Promote your content on social media by automatically publishing when you publish on your site.', 'jetpack-my-jetpack' );
+	}
+
+	/**
+	 * Get the internationalized features list
+	 *
+	 * @return array Boost features list
+	 */
+	public static function get_features() {
+		return array(
+			__( 'Post to social networks', 'jetpack-my-jetpack' ),
+			__( 'Schedule publishing', 'jetpack-my-jetpack' ),
+			__( 'Supports the major social networks', 'jetpack-my-jetpack' ),
+		);
+	}
+
+	/**
+	 * Get the product princing details
+	 *
+	 * @return array Pricing details
+	 */
+	public static function get_pricing_for_ui() {
+		return array_merge(
+			array(
+				'available'          => true,
+				'wpcom_product_slug' => static::get_wpcom_product_slug(),
+			),
+			Wpcom_Products::get_product_pricing( static::get_wpcom_product_slug() )
+		);
+	}
+
+	/**
+	 * Get the WPCOM product slug used to make the purchase
+	 *
+	 * @return ?string
```

Comment:
Is this supposed to be `?string`? Looks like it will never be null.

Reply:
It won't in this instance, but this is the return type for the function this overrides, which by default returns `null`. It looks like the other product classes have kept this definition too. It makes me wonder if we need this docblock at all!

Reply:
For these functions, I have updated the return type to string, since that's what we are returning.